test/fn/watch-project.spec.js

```diff
@@ -0,0 +1,223 @@
+    fs.fs.copyFileSync(path.join(absSampleDir, file), path.join(testDir, destination, file));
+  });
+}
+
+function watchWrapper(action, file) {
+  return new Promise((resolve,) => {
+    watchProject.watch(mockApi, action, async (path) => {
+      if (path === file) {
+        resolve();
+      }
+    });
+  });
+}
+
+describe('watch-project', function () {
+
+  beforeEach(() => {
+    sinon.stub(environment, 'pathToProject').get(() => testDir);
+    sinon.stub(environment, 'extraArgs').get(() => { });
+    sinon.stub(environment, 'isArchiveMode').get(() => false);
+    sinon.stub(environment, 'skipTranslationCheck').get(() => false);
+    sinon.stub(environment, 'force').get(() => false);
+    return api.db.put({ _id: '_design/medic-client', deploy_info: { version: '3.5.0' } }).then(() => api.start());
+  });
+
+  afterEach(() => {
+    sinon.restore();
+    fs.writeJson(settingsPath, appSettings);
+    watchProject.close();
+    return api.stop();
+  });
+
+  it('watch-project: upload app settings', () => {
+    return watchWrapper(editAppSettings, 'app_settings.json')
+      .then(mockApi.getAppSettings)
+      .then((settings) => { return JSON.parse(settings.content); })
+      .then((settings) => expect(settings.locale).equal('es'));
+  });
+
+  it('watch-project: convert app settings', () => {
+    const appSettingsPath = path.join(testDir, 'app_settings.json');
+    fs.fs.unlinkSync(appSettingsPath);
+    expect(fs.fs.existsSync(appSettingsPath)).to.be.false;
+    return watchWrapper(editBaseSettings, 'base_settings.json')
+      .then(() => { return fs.readJson(appSettingsPath); })
+      .then((settings) => expect(settings.locale).equal('es'));
+  });
+
+  it('watch-project: upload custom translations', () => {
+    return uploadCustomTranslations()
+      .then(() => expectTranslationDocs(api, 'en'))
+      .then(watchWrapper(editTranslations, 'messages-en.properties'))
+      .then(() => getTranslationDoc(api, 'en'))
+      .then(messages => {
+        assert.deepEqual(messages.custom, { a: 'first', test: 'new' });
+      });
+  });
+
+  it('watch-project: upload resources', () => {
+    return watchWrapper(editResources, 'resources.json')
+      .then(() => api.db.allDocs())
+      .then(docs => {
+        expect(docs.rows.filter(row => row.id === 'resources')).to.not.be.empty;
+      })
+      .then(() => {
+        fs.writeJson(resourceJsonPath, {});
+      });
+  });
+
+  it('watch-project: convert app forms', () => {
+    const appFormPath = path.join(testDir, 'forms', 'app');
+    const copyForm = () => copySampleForms('convert-app-form');
+    return watchWrapper(copyForm, 'death.xlsx')
+      .then(() => {
+        expect(fs.fs.readdirSync(appFormPath).find(file => file === 'death.xml')).to.be.not.undefined;
+      })
+      .then(() => {
+        fs.fs.readdirSync(appFormPath).filter(name => name.startsWith('death')).forEach(file => fs.fs.unlinkSync(path.join(appFormPath, file)));
```

Comment:
Please consider saving `death` string as a variable and reuse it. It's used 3 times in this test.